glfs.c

```diff
@@ -412,7 +412,7 @@ static glfs_t* tcmu_create_glfs_object(char *config, gluster_server **hosts)
 	gluster_cache_refresh(fs, config);
 
  fail:
-	gluster_free_server(&entry);
+	gluster_free_server(hosts);
```

Comment:
I found a similar bug even in parse_imagepath() as well. Will update the pull request I sent.